types/src/nibble/nibble_path/mod.rs

```diff
@@ -104,6 +108,32 @@ impl NibblePath {
         NibblePath { num_nibbles, bytes }
     }
 
+    pub fn new_from_state_key(state_key: &StateKey, num_nibbles: usize) -> Self {
+        NibblePath::new_from_byte_array(state_key.hash().to_vec(), num_nibbles)
+    }
+
+    fn new_from_byte_array(mut bytes: Vec<u8>, num_nibbles: usize) -> Self {
+        checked_precondition!(num_nibbles <= ROOT_NIBBLE_HEIGHT);
+        if num_nibbles % 2 == 1 {
+            // Rounded up number of bytes to be considered
+            let num_bytes = (num_nibbles + 1) / 2;
+            checked_precondition!(bytes.len() >= num_bytes);
+            // If number of nibbles is odd, make sure to pad the last nibble with 0s.
+            let last_byte_padded = bytes[num_bytes - 1] & 0xF0;
+            bytes[num_bytes - 1] = last_byte_padded;
+            NibblePath {
+                num_nibbles,
+                bytes: bytes[..num_bytes].to_vec(),
```

Comment:
nit: second time we memcopy, should avoid

Reply:
Fixed to have only one memcopy.